packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -244,6 +245,21 @@ export interface ClusterProps extends ClusterOptions {
    */
   readonly kubectlEnabled?: boolean;
 
+  /**
+   * Enable or disable exporting the Kubernetes control plane logs for your cluster to CloudWatch Logs.
+   *
+   * @default false Cluster control plane logs are not exported to CloudWatch Logs.
+   */
```

Comment:
What happens if I enabled logs and then set to undefined (remove it). Would it set all the logging to false?

Reply:
it will send an empty value for logging in the call to `updateClusterConfig()`, and it would result in setting the default value from AWS

Reply:
Have you tested this? My intuition is that if I send an `updateClusterConfig` request with an empty value for `logging` it will not perform any changes to my logging configuration, which is *not* what users expect if they enable logs and then remove this property in their CDK app.

I suspect that we may either need to *always* send the two enabled/disable lists *or* add heuristics to the UPDATE handler to calculate the "diff" from the previous value and produce an appropriate update.

Reply:
i will try this out and let you know, but this diff is being done in the runtime code here: https://github.com/aws/aws-cdk/blob/53d647f25d5ad92817eb8d0773b5d1af9b95a96a/packages/%40aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts#L293

Reply:
This only determines that there was a change, not which log types should now be enabled/disabled based on that change.

For example, say, "oldProps" included two enabled types (say `["typeA", "typeB"]`) and now "newProps" include only a single type (say `["typeB"]`). Then, the request we need to send to `updateClusterConfig` is only to __disable__ (we can, if we want also include an "enable" for `typeB`, but technically it's not required because it is already enabled).

LMK if this makes sense.

Reply:
i believe we will need to send the whole configuration explicitly, including the disabled one. i am getting the following error when sending the exact same configuration to `updateClusterConfig`:
> Failed to update resource. Error: No changes needed for the logging config provided